src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -521,8 +520,29 @@ public List<EmailWrapper> generateFeedbackSessionClosingWithExtensionEmails(
         }
 
         String template = EmailTemplates.USER_FEEDBACK_SESSION.replace("${status}", FEEDBACK_STATUS_SESSION_CLOSING);
-        return generateFeedbackSessionEmailBases(course, session, students, instructors, Collections.emptyList(), template,
-                EmailType.FEEDBACK_CLOSING, FEEDBACK_ACTION_SUBMIT_EDIT_OR_VIEW);
+        StringBuilder studentAdditionalContactBuilder = new StringBuilder();
+        StringBuilder instructorAdditionalContactBuilder = new StringBuilder();
+        studentAdditionalContactBuilder
+                .append(HTML_NO_ACTION_REQUIRED)
+                .append(getAdditionalContactInformationFragment(course, false));
+        instructorAdditionalContactBuilder
+                .append(HTML_NO_ACTION_REQUIRED)
+                .append(getAdditionalContactInformationFragment(course, true));
+
+        List<EmailWrapper> emails = new ArrayList<>();
+        EmailType type = EmailType.FEEDBACK_CLOSING;
+        String feedbackAction = FEEDBACK_ACTION_SUBMIT_EDIT_OR_VIEW;
+        for (StudentAttributes student : students) {
+            FeedbackSessionAttributes studentSession = session.getCopyForStudent(student.getEmail());
+            emails.add(generateFeedbackSessionEmailBaseForStudents(course, studentSession, student,
+                    template, type, feedbackAction, studentAdditionalContactBuilder.toString()));
+        }
+        for (InstructorAttributes instructor : instructors) {
+            FeedbackSessionAttributes instructorSession = session.getCopyForInstructor(instructor.getEmail());
+            emails.add(generateFeedbackSessionEmailBaseForInstructors(course, instructorSession, instructor,
+                    template, type, feedbackAction, instructorAdditionalContactBuilder.toString()));
+        }
```

Comment:
Does this work:
```java
for (StudentAttributes student : students) {
    FeedbackSessionAttributes studentSession = session.getCopyForStudent(student.getEmail());
    emails.add(generateFeedbackSessionEmailBases(course, studentSession, Collections.singletonList(student),
            new ArrayList<>(), new ArrayList<>(), template, type, feedbackAction));
}
for (InstructorAttributes instructor : instructors) {
    FeedbackSessionAttributes instructorSession = session.getCopyForInstructor(instructor.getEmail());
    emails.add(generateFeedbackSessionEmailBases(course, instructorSession, new ArrayList<>(),
            Collections.singletonList(instructor), new ArrayList<>(), template, type, feedbackAction));
}
```
i.e. no need to repeat the additional contact builders

Reply:
Yes this will work too. Will make the change.

Reply:
Changed to the following based on your suggestion:
https://github.com/TEAMMATES/teammates/blob/d74d1500092a6ae21fe670016cbb3478de2a0d0b/src/main/java/teammates/logic/api/EmailGenerator.java#L527-L536